packages/ui/molecules/Menu/Menu.js

```diff
@@ -0,0 +1,241 @@
+import { Base } from '@studiometa/js-toolkit';
+import { nextTick } from '@studiometa/js-toolkit/utils';
+import MenuBtn from './MenuBtn.js';
+import MenuItems from './MenuItems.js';
+
+/**
+ * @typedef {Menu & {
+ *   $children: {
+ *     Menu: Menu[],
+ *     MenuBtn: MenuBtn[],
+ *     MenuItems: MenuItems[],
+ *   }
+ * }} MenuInterface
+ */
+
+/**
+ * Menu class.
+ */
+export default class Menu extends Base {
+  /**
+   * Config.
+   */
+  static config = {
+    name: 'Menu',
+    debug: true,
+    components: {
+      MenuBtn,
+      MenuItems,
+      Menu,
+    },
+    options: {
+      mode: {
+        type: String,
+        default: 'click', // or 'hover'
```

Comment:
**question/improvement**: Does the mode is a strict  choice between `click` & `hover` ? 
This behavior can be link to the device, on phone/tablet the behavior needs to be click, be could be hover on computers, for a mega menu like the demo

Reply:
I think that we can keep the option logic with `click` & `hover` and override the `shouldReactOnClick` getter in a custom menu if needed. 

We can decide that a mega-menu on desktop should react on hover and its mobile equivalent in a burger menu should react on click:  

```js
get shouldReactOnClick() {
  return ['xxs', 'xs', 's'].includes(this.$services.get('resized').breakpoint);
}
```